framework/decode/vulkan_rebind_allocator.cpp

```diff
@@ -794,67 +832,99 @@ VkResult VulkanRebindAllocator::BindImageMemory2(uint32_t                     bi
-                auto          resource_alloc_info = reinterpret_cast<ResourceAllocInfo*>(allocator_image_data);
-                auto          memory_alloc_info   = reinterpret_cast<MemoryAllocInfo*>(allocator_memory_data);
-
-                VkMemoryRequirements requirements;
-                functions_.get_image_memory_requirements(device_, image, &requirements);
-
-                VmaAllocationCreateInfo create_info;
-                create_info.flags = 0;
-                create_info.usage = GetImageMemoryUsage(
-                    resource_alloc_info->usage,
-                    resource_alloc_info->tiling,
-                    capture_memory_properties_.memoryTypes[memory_alloc_info->original_index].propertyFlags,
-                    requirements);
-                create_info.requiredFlags  = 0;
-                create_info.preferredFlags = 0;
-                create_info.memoryTypeBits = 0;
-                create_info.pool           = VK_NULL_HANDLE;
-                create_info.pUserData      = nullptr;
+                auto memory_alloc_info = reinterpret_cast<MemoryAllocInfo*>(allocator_memory_data);
+                if (memory_alloc_info->ahb)
+                {
+                    VkDeviceMemory memory        = bind_infos[i].memory;
+                    VkDeviceSize   memory_offset = bind_infos[i].memoryOffset;
+
+                    VkImportAndroidHardwareBufferInfoANDROID importAHBInfo;
+                    importAHBInfo.sType  = VK_STRUCTURE_TYPE_IMPORT_ANDROID_HARDWARE_BUFFER_INFO_ANDROID;
+                    importAHBInfo.pNext  = nullptr;
+                    importAHBInfo.buffer = memory_alloc_info->ahb;
+
+                    VkMemoryDedicatedAllocateInfo dedicatedAllocateInfo{};
+                    dedicatedAllocateInfo.sType = VK_STRUCTURE_TYPE_MEMORY_DEDICATED_ALLOCATE_INFO;
+                    dedicatedAllocateInfo.pNext = &importAHBInfo;
+                    dedicatedAllocateInfo.image = image;
+
+                    VkMemoryAllocateInfo allocate_info{};
+                    allocate_info.sType           = VK_STRUCTURE_TYPE_MEMORY_ALLOCATE_INFO;
+                    allocate_info.pNext           = &dedicatedAllocateInfo;
+                    allocate_info.allocationSize  = memory_alloc_info->allocation_size;
+                    allocate_info.memoryTypeIndex = memory_alloc_info->original_index;
+                    result = functions_.allocate_memory(device_, &allocate_info, nullptr, &memory);
 
-                VmaAllocationInfo allocation_info;
-                result = vmaAllocateMemoryForImage(allocator_, image, &create_info, &allocation, &allocation_info);
+                    if (result >= 0)
+                    {
+                        assert(allocate_info.memoryTypeIndex < replay_memory_properties_.memoryTypeCount);
 
-                if (result >= 0)
+                        result = functions_.bind_image_memory(device_, image, memory, memory_offset);
+                    }
```

Comment:
This block looks duplicated from BindImageMemory, does it make sense to put it in a helper?